modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/ProposalTrustSelect.scala

```diff
@@ -19,19 +18,18 @@ import eu.timepit.refined.numeric.Positive
 object ProposalTrustSelect {
 
   def make[F[_]: MonadThrow](
-    getTrusts: F[(Option[Map[PeerId, Double]], TrustMap)],
+    getTrusts: F[(TrustLabels, TrustScores)],
     config: ProposalSelectConfig
   ): ProposalSelect[F] = (declarations: Map[PeerId, PeerDeclarations]) =>
     for {
       (d, r) <- getTrusts
-      deterministic = d
-        .map(_.filter { case (peerId, s) => s > 0 && declarations.contains(peerId) })
-        .getOrElse(Map.empty)
-
-      relative = r.trust.view
-        .mapValues(_.predictedTrust)
-        .collect { case (peerId, Some(s)) if s > 0 && declarations.contains(peerId) => peerId -> s }
-        .toMap
+      deterministic = d.value.filter {
+        case (peerId, s) => s > 0 && declarations.contains(peerId)
+      }
+
+      relative = r.value.collect {
+        case (peerId, s) if s > 0.0 && declarations.contains(peerId) => peerId -> s
```

Comment:
Isn't this also a "filter" ?
If so you can make it a function and clean up its two uses.

Reply:
Refactored to

```
      proposalSelect =
        if (cfg.environment === Dev)
          ProposalOccurrenceSelect.make()
        else {
          val getTrusts = Applicative[F].product(
            p2PClient.l0Trust.getPreviousSeedlistPeerLabels.run(cfg.globalL0Peer),
            p2PClient.l0Trust.getCurrentTrust.run(cfg.globalL0Peer)
          )

          val primary = ProposalTrustSelect.make(getTrusts, cfg.proposalSelect)
          val secondary = ProposalOccurrenceSelect.make()

          ProposalSelectWithFallback.make(primary, secondary)
        }
```